### PR TITLE
do not check for homebrew is packages present

### LINKED ISSUE #16458: flutter should not require brew on mac
I am assuming I an not the only person in the world who is happy using Macports, and do not require brew. (In fact using both creates conflicts). 

I have been perfectly able to work with Android, iOS, NodeJS development, including Nativescript and React Native over the last 5 years, without ever requiring brew. 

I have been able to install, including ideviceinstaller, ios-deploy and cocoapods without brew. Why does flutter complain about brew, when the required packages are already present.

### PR DESCRIPTION
**Pull request by championswimmer**
if cocoapods, ideviceinstaller and ios-deploy are installed and working fine, we do not need homebrew at all

resolves #16458

### CODE REVIEW

**Comment by dnfield** [comment]
Looks like CI is failing because you're missing a semicolon on line 128.

File: packages/flutter_tools/lib/src/ios/ios_workflow.dart [hunk 1]
```diff
@@ -120,9 +120,18 @@ class IOSWorkflow extends DoctorValidator implements Workflow {
       pythonStatus = ValidationType.missing;
       messages.add(new ValidationMessage.error(kPythonSix.errorMessage));
     }
+    bool requireHomeBrew = true;
+    
+    if (iMobileDevice.isInstalled && (await iMobileDevice.isWorking) && (await hasIDeviceInstaller)) {
```
Comment (xster) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: We might want to now cache these values with these repeated checks

**Review by xster — COMMENTED** [review]
Can you add some tests to ios_workflow_test as well?

**Comment by championswimmer** [comment]
I'll try to write the tests, but I have no idea to be honest. I just wrote my first line of Dart yesterday :)

**Comment by xster** [comment]
Sure, check out ios_workflow_test.dart. You can copy paste one of the existing test. The mocking should be somewhat self explanatory. You can mock out the new conditions that matches your `if`s but have `hasHomebrew` false and see that there are no error messages.

**Comment by Hixie** [comment]
@championswimmer  I'm going to close this PR since you're not working on it right now, but if you have time to come back to it please don't hesitate to reopen or open a new one or whatever, and if you need anything please don't hesitate to cc me. Thanks!

**Comment by goderbauer** [comment]
@jslavitz Can we close this?

**Pull request closed without merging**